src/main/java/rs/teslaris/core/service/impl/DocumentFileServiceImpl.java

```diff
@@ -0,0 +1,205 @@
+            titleLanguageDetected, documentFile, serverFilename, documentIndex);
+
+        documentFileIndexRepository.save(documentIndex);
+    }
+
+    private boolean isPdfFile(MultipartFile multipartFile) {
+        return Objects.equals(multipartFile.getContentType(), "application/pdf");
+    }
+
+    private String extractDocumentContent(MultipartFile multipartPdfFile) {
+        String documentContent;
+        try (var pdfFile = multipartPdfFile.getInputStream()) {
+            var pdDocument = PDDocument.load(pdfFile);
+            var textStripper = new PDFTextStripper();
+            documentContent = textStripper.getText(pdDocument);
+            pdDocument.close();
+        } catch (IOException e) {
+            throw new LoadingException("Error while trying to load PDF file content.");
+        }
+        return documentContent;
+    }
+
+    private String extractDocumentTitle(MultipartFile multipartPdfFile) {
+        var originalFilename = Objects.requireNonNull(multipartPdfFile.getOriginalFilename());
+        return originalFilename.split("\\.")[0];
+    }
+
+    private String detectLanguage(String text) {
+        LanguageDetector languageDetector;
+        try {
+            languageDetector = LanguageDetector.getDefaultLanguageDetector().loadModels();
+        } catch (IOException e) {
+            throw new NotFoundException("Error while loading language models.");
+        }
+        return languageDetector.detect(text).getLanguage();
+    }
+
+    private void saveDocumentIndex(String documentContent, String documentTitle,
+                                   String contentLanguageDetected, String titleLanguageDetected,
+                                   DocumentFile documentFile, String serverFilename,
+                                   DocumentFileIndex documentIndex) {
+
+        if (contentLanguageDetected.equals("hr") || contentLanguageDetected.equals("sr")) {
+            documentIndex.setPdfTextSrp(documentContent);
+        } else {
+            documentIndex.setPdfTextOther(documentContent);
+        }
+
+        if (titleLanguageDetected.equals("hr") || titleLanguageDetected.equals("sr")) {
+            documentIndex.setTitleSrp(documentTitle);
+        } else {
+            documentIndex.setTitleOther(documentTitle);
+        }
+
+        documentIndex.setDescriptionSrp("");
+        documentFile.getDescription().stream()
+            .filter(d -> d.getLanguage().getLanguageTag().startsWith("SR")).forEach(
+                d -> documentIndex.setDescriptionSrp(
+                    documentIndex.getDescriptionSrp() + d.getContent()));
+
+        documentIndex.setDescriptionOther("");
+        documentFile.getDescription().stream()
+            .filter(d -> !d.getLanguage().getLanguageTag().startsWith("SR")).forEach(
+                d -> documentIndex.setDescriptionOther(
+                    documentIndex.getDescriptionOther() + d.getContent() + " | "));
```

Comment:
why description is not handled in the same way as title and PdfText? Do we need besides contentLanguageDetected and titleLanguageDetected, also descriptionLanguageDetected?

Reply:
Because it is MultilingualContent and as such, the language for it is provided when submitting the form